src/ntimporters/utils.py

```diff
@@ -80,14 +82,97 @@ def strip_readonly(model: ModelNormal):
     )
-    group_id = st_groups[0].get("id") if st_groups and st_groups[0] else None
+    return str(st_groups[0].get("id")) if st_groups and st_groups[0] else None
+
+
+def exists(entity_type: str, name: str, imported_entities: dict[str, tuple[str, str]]) -> dict:
+    """Check if entity already exists and return its id"""
+    if imported_entities:
+        if entity_type == "comments":
+            name = name[:10]
+        if (records := imported_entities.get(entity_type)) and (record := records.get(name)):
+            return record
+    return Dict({"id": None})
+
+
+class Dict(UserDict):
+    """Class pretending OpenApi object and dict in the same time"""
+
+    @property
+    def id(self):
+        """Id as a property"""
+        return self.get("id")
+
+    def __len__(self):
+        """Check if none"""
+        return self.get("id") is not None
+
+
+def get_imported_entities(nt_client, team_id, group_name) -> dict[str, list]:
+    """Get already imported records"""
+    already_imported = []
+    if group_id := get_group_id(nt_client, team_id, group_name):
+        for pgroup in _get_with_query(
+            nt_client,
+            apis.GroupAssignmentsApi(nt_client).get_group_assignments_endpoint,
+            [("group_id", group_id), ("group_type", "project")],
+        ):
+            project = apis.ProjectsApi(nt_client).get_project_by_id(str(pgroup.get("object_id")))
+            already_imported.append(("project", project))
+            for section in _get_with_query(
+                nt_client,
+                apis.ProjectSectionsApi(nt_client).get_project_sections_endpoint,
+                [("project_id", str(project.get("id")))],
+            ):
+                already_imported.append(("project_section", section))
+            for task in _get_with_query(
+                nt_client,
+                apis.TasksApi(nt_client).get_tasks_endpoint,
+                [("project_id", str(project.get("id")))],
+            ):
+                already_imported.append(("task", task))
+                for comment in _get_with_query(
+                    nt_client,
+                    apis.CommentsApi(nt_client).get_comments_endpoint,
+                    [("task_id", str(task.get("id")))],
+                ):
+                    already_imported.append(("comment", comment))
+                for tag in _get_with_query(
+                    nt_client,
+                    apis.TagsApi(nt_client).get_tags_endpoint,
+                    [("task_id", str(task.get("id")))],
+                ):
+                    already_imported.append(("tag", tag))
+    entities = {
+        "comments": {
+            str(elt[1].get("body"))[:10]: Dict({"id": elt[1].get("id")})
+            for elt in already_imported
+            if elt[0] == "comment"
+        }
+    }
+    for rtype in ("task", "tag", "project", "project_section"):
+        entities[f"{rtype}s"] = {
+            str(elt[1].get("name")): Dict({"id": elt[1].get("id")})
```

Comment:
for now it's ok, but we could improve it by preserving the hierarchy of projects/tasks/comments. Otherwise if we have e.g. 2 tasks with the same name in 2 different projects and only one was migrated, the other one won't get migrated due to name conflict

Reply:
I believe i wrote almost same comment somewhere related to `np.py` ;-). Kosher way would be to simply track 3rd party id's and commit into some temporary record, however it would need extension of API.